106 neighbourhood ses variables

loffleraSMH opened the pull request:
This PR adds a new function to derive neighbourhood-level SES variables from the Statcan census data and Ontario Marginalization Index. 

- The function currently only returns variables from the `statcan` table that have been used by several projects in the past (as discussed during sprint in May 2024)
- The function documentation provides information about the relevant data sources and how they were linked to GEMINI encounters, and also provides a brief description of all returned variables with links to relevant resources
- The function queries the relevant GEMINI tables, calculates the outcome variables (e.g., % visible minorities), and replaces missing/invalid values with NA (showing warning messages missing values)
- I tested the function on both 2016 and 2021 Statcan data and made sure the function returns error messages in case users select tables that aren't available (e.g., `drm_cleandb_v2`  only contains 2016 census data)
- I also checked the output to make sure all variables are returned as expected and only include plausible values (e.g., all entries where quintile = 9 have been set to NA)

For the reviewer: Please review the function documentation for clarity & completeness, and also test the function on different DB versions.

You can use the following code as a starting point (please note that you need to source  `utils.R` from branch 106 for an updated version of `find_db_tablename()`, which can identify the relevant `lookup_statcan` table:

```
library(DBI)
library(RPostgreSQL)
library(getPass)
library(dplyr)
library(data.table)

drv <- dbDriver("PostgreSQL")
db <- DBI::dbConnect(drv,
    dbname = "drm_cleandb_v3_1_0",
    host = "prime.smh.gemini-hpc.ca",
    port = 5432,
    user = getPass("User:"),
    password = getPass("Enter Password:")
)

# change this to your Rgemini folder and pull from branch 106
source("~/GitHub/Rgemini/R/utils.R")
source("~/GitHub/Rgemini/R/neighborhood_ses.R")

cohort <- dbGetQuery(db, "select genc_id from admdad LIMIT 1000") %>% data.table() 

# get neighbourhood-level variables based on 2021 statcan data
census <- neighbourhood_ses(db, cohort, 2021)

# for 2016 census
# note: both neighbourhood_ses() and neighborhood_ses() should work since I created an alias
census <- neighborhood_ses(db, cohort, 2016)
```

lamphy-smh commented on R/neighbourhood_ses.R:
```diff
@@ -0,0 +1,302 @@
+#' @section Missing values:
+#' Some encounters could not be linked to Statistics Canada data due to missing/
+#' invalid postal codes, or due to the fact that they reside in an area not
+#' covereded by the census. These encounters will be returned with `da_uid = NA`.
```
slight typo here on 'covered'

loffleraSMH replied:
Thanks, I fixed the typo.

lamphy-smh commented on R/neighbourhood_ses.R:
```diff
@@ -0,0 +1,302 @@
+#' language and better reflect the census measures associated with each dimension.
+#'
+#' All ON-Marg variables are available as continuous factor scores as well as
+#' quintiles. Higher scores represent a higher degree of marginalization.
```
Don't know if you need to indicate negative scores are possible for ON-Marg?

lamphy-smh commented on _pkgdown.yml:
```diff
@@ -31,9 +31,11 @@ reference:
   - readmission
   - season
   - stat_holidays_ON
+  - neighbourhood_ses
```
`neighbourhood_ses` is listed twice?

loffleraSMH replied:
Good catch, I removed the duplicate entry.

lamphy-smh reviewed (commented):
Tested function and it works great! 
Just couple of comments/suggestions on the files and roxygen documentation.

loffleraSMH commented:
> Tested function and it works great! Just couple of comments/suggestions on the files and roxygen documentation.

Thanks for the review @lamphy-smh! I addressed your comments and will go ahead and merge this into `develop`.
